webapp/src/app/admin/devices/devices.component.ts

```diff
@@ -84,11 +95,40 @@ export class DevicesComponent implements OnInit {
                         });
                 const url = window.location.protocol + "//" + window.location.host +
                     this.location.prepareExternalUrl(device);
-                this.dialog.open(DeviceShowComponent, {data: url});
+                this.dialog.open(ShowComponent, {data: url});
             }
         });
     }
 
+    async renameDevice(dev: Device, typeStr: string) {
```

Comment:
`typeStr` is used mainly to check if it is handling a recovery device or not. Maybe renaming it to `isRecovery: boolean` to show that and avoid external invalid calls.

Reply:
Doesn't look like a boolean to me. It could be an `enum`.

Reply:
You kept a string type; at least, use a [string literal type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#string-literal-types) if you really want to use the string valuespace.